### PR TITLE
Use default ssh key if already exists in snow devices

### PR DESCRIPTION
**Pull request by jiayiwang7**
*Issue #, if available:*

Currently we create a default ssh key with name `eksa-default` and import it to snow devices. 
There is a race. If multiple users create clusters in the same device set at the same time, eks-a creates default key for first cluster. The second cluster validation will fail at ec2 import-key-pair step because the key already exists in devices.

*Description of changes:*

Before creating default ssh key, we check if the key exists on devices first. If the key exists on all devices, we skip the key generation and only assign the key name to machine config.

*Testing (if applicable):*

- unit tests

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: pkg/providers/snow/defaults.go [hunk 1]
```diff
@@ -70,6 +84,21 @@ func (md *MachineConfigDefaulters) SetupDefaultSshKey(ctx context.Context, m *v1
 		return nil
 	}
 
+	keyCount, err := md.defaultKeyCount(ctx)
+	if err != nil {
+		return err
+	}
+
+	if keyCount > 0 && keyCount < len(md.defaulters.awsClientMap) {
+		return fmt.Errorf("default key [keyName=%s] only exists on some of the devices. Use 'aws ec2 import-key-pair' to import this key to all the devices", defaultAwsSshKeyName)
```
Comment (g-gaston) [inline pkg/providers/snow/defaults.go:93]: nit: shouldn't we ideally just import it wherever is needed? I'm ok if you want that as an improvement later

File: pkg/providers/snow/defaults.go [hunk 2]
```diff
@@ -70,6 +84,21 @@ func (md *MachineConfigDefaulters) SetupDefaultSshKey(ctx context.Context, m *v1
...
+	}
+
+	if keyCount == len(md.defaulters.awsClientMap) {
```
Comment (g-gaston) [inline pkg/providers/snow/defaults.go:96]: do we need a test case for this scenario?

File: pkg/providers/snow/defaults.go (hunk 1 above)
Comment (jiayiwang7) [reply pkg/providers/snow/defaults.go:93]: we don't have access to the public key tho. Thus we can't import for the user

Comment (jiayiwang7) [reply pkg/providers/snow/defaults.go:93]: will follow up on this

**Review by g-gaston — APPROVED** [review]
/lgtm

**Pull request merged**